kele.el

```diff
@@ -1728,120 +1734,47 @@ Otherwise, simply `kele-get' the resource at point."
-             ("Restarts" . (lambda (r)
-                             (let-alist r
-                               (->> .status.containerStatuses
-                                    (-map (lambda (status)
-                                            (alist-get 'restartCount status)))
-                                    (-sum)))))))
-    (deployments . (("Ready" . (lambda (r)
-                                 (let-alist r (format "%s/%s" .status.readyReplicas .status.replicas))))
-                    ("Up-to-date" . (lambda (r)
-                                      (let-alist r .status.updatedReplicas)))
-                    ("Available" . (lambda (r) (let-alist r
-                                                 .status.readyReplicas))))))
-  "Alist containing column specifications for `kele-list'.
-
-Keys are symbols representing the plural form of Kubernetes
-resource kinds, e.g. `deployments'.  Values are alists mapping
-column names to unary functions that take the resource object and
-return the corresponding value.
-
-The nil key contains columns general to all resources.")
-
-(defun kele--make-list-vtable (gvk context namespace)
+(defun kele--vtable-tabulate (gvk context namespace)
   "Construct an interactive vtable listing resources of GVK.
 
 CONTEXT and NAMESPACE are according to Kubernetes conventions and
 serve to further specify the resources to list.
 
-If NAMESPACE is nil, displays resources for all namespaces."
-  (let ((columns
-         (append
-          ;; FIXME: Pull these from the nil-key entries in kele--list-columns
-          '((:name "Name" :width 30 :align left)
-            (:name "Namespace" :width 20 :align left)
-            (:name "GVK" :width 10 :align left)
-            (:name "Owner(s)" :width 20 :align left)
-            (:name "Created" :width 30 :align left))
-          (mapcar (lambda (key)
-                    `(:name ,key :width ,(length key) :align left))
-                  (map-keys (alist-get (intern (oref gvk kind)) kele--list-columns))))))
+If NAMESPACE is nil, displays resources for all namespaces.
+
+Uses server rendering to retrieve the columns and their values.  For more
+details, see:
+https://kubernetes.io/docs/reference/using-api/api-concepts/#receiving-resources-as-tables."
+  (let* ((resource-table (kele--tabulate-resources
+                          gvk
+                          :context context
+                          :namespace namespace))
+         (colnames (-map (lambda (coldef)
+                           (let-alist coldef .name))
+                         (-filter (lambda (coldef)
+                                    (let-alist coldef (= .priority 0)))
+                                  (append (alist-get 'columnDefinitions
+                                                     resource-table)
+                                          '()))))
+         (column-specs (-map (lambda (colname)
+                               `(:name ,colname :width 30 :align left))
```

Comment:
We have access to all of the values for each column, which means that we can set the width to be slightly larger than the max length. Let's do that.